index.rst

```diff
@@ -1,51 +1,1166 @@
+Additionally, the while we strive to have different developers work as much as possible on independent packages, the numerous
+interdependencies will sometimes require working on the same package and even in the same ticket branch.
+Thus frequent pushing and rebasing is necessary to keep everyones stack up to date.
+To prevent unnecessary merge conflicts it is best to lock files that you are currently working on.
+Git does not have a true lock, in that locking a file does not prevent others from working on it and pushing their commits.
+Instead the `gitlock`_ package can be used to lock a particular file and notify the group that a file is being worked on. Once gitlock is setup you can lock a file by using
+
+.. code-block:: bash
+
+    gitlock lock {{package name}} -f <relative path to the file>
+
+The script will notify you if you were able to successfully lock the file or if it is already locked by another user. Once you have finished working on a file, using
+
+.. code-block:: bash
+
+    gitlock unlock {{package name}} -f <relative path to the file>
+
+will unlock the file and allow others to work on it.
+
+.. warning::
+
+    Remember that gitlocks do not prevent you or other users from modifying files and committing changes.
+    Do your best to be considerate of other developers and try to lock and unlock files as needed.
+
+.. _activate_test:
+
+Activate the Test
+-----------------
+
+All of the tests that have yet to be wrapped are commented out in a file called ``tests/test.txt`` (currently only in ``afw``).
+To start wrapping the code for a test uncomment it, run it and see what breaks.
+For tests with multiple test classes and methods it may be useful to add the decorator ``@unittest.skip("pybind11 temporary skip while wrapping")`` to functions in a test that have not been wrapped yet.
+
+Tutorial
+========
+
+To illustrate how to wrap a test we will use ``afw/tests/testMinimize.py`` as an example. We start by cloning https://github.com/lsst/afw to our local machine and checkout the correct ticket branch for the current test. In this case ``testMinimize.py`` is in ``tickets/DM-6298``, so we checkout that branch and set it up with ``setup -r .`` from the main ``afw`` repository directory.
+
+Compiling the Code
+------------------
+
+Before we make any changes it's a good idea to compile the cloned repository to make sure that everything is setup correctly. From the ``afw`` repository main directory run
+
+.. code-block:: shell
+
+    git clean -dfx
+
+followed by
+
+.. code-block:: shell
+
+    scons lib python
+
+to do a clean build of afw.
+Since this is your first build of afw it will take a while but as you make changes, using
+
+.. code-block:: shell
+
+    scons lib python
+
+only builds the newly wrapped headers, so development is much faster than with SWIG).
+
+Locking Files
+-------------
+
+Before we start working we want to lock the current test using
+
+.. code-block:: bash
+
+    gitlock lock afw -f tests/testMinimize.py
```

Comment:
No, that's because some tests were renamed/added after we created the initial gitlock file. There is a command that will search the directory tree and update the file, perhaps that should happen automatically if the test isn't found, but I guess no one ran that command when a test wasn't found.